test/lift.lean

```diff
@@ -189,3 +190,21 @@ example (x : WithBot Unit) (h : x ≠ ⊥) : True := by
   guard_hyp x : Unit
   guard_hyp h : (x : WithBot Unit) ≠ ⊥
   trivial
+
+example (n : ℕ) (hn : 0 < n) : True := by
+  lift n to ℕ+
+  · guard_target =ₛ 0 < n
+    exact hn
+
+  guard_hyp n : ℕ+
```

Comment:
Without the `PNat.val` change, this fails with `hypothesis n has type { n // 0 < n }, not ℕ+`